visio_gui.py

```
@@ -0,0 +1,537 @@
+                
+            # Parse JSON results
+            try:
+                files_data = json.loads(result.stdout.strip())
+                # Handle both single object and array cases
+                if isinstance(files_data, dict):
+                    files_data = [files_data]
+                elif not isinstance(files_data, list):
+                    files_data = []
+                self.root.after(0, self._scan_complete, files_data)
+            except json.JSONDecodeError as e:
+                self.root.after(0, lambda: messagebox.showerror("Error", f"Error parsing scan results: {str(e)}"))
```

Ruff (0.12.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)